pkg/handlers/generic/lifecycle/utils/utils.go

```diff
@@ -6,17 +6,38 @@ package utils
 import (
 	"context"
 	"fmt"
+	"maps"
 
 	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/remote"
 	crsv1 "sigs.k8s.io/cluster-api/exp/addons/api/v1beta1"
+	capiutil "sigs.k8s.io/cluster-api/util"
 	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
 	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/k8s/client"
 )
 
+const (
+	kindStorageClass = "StorageClass"
+)
+
+var (
+	defualtStorageClassKey = "storageclass.kubernetes.io/is-default-class"
+	defaultStorageClassMap = map[string]string{
+		defualtStorageClassKey: "true",
+	}
+	defaultParams = map[string]string{
+		"csi.storage.k8s.io/fstype": "ext4",
+		"type":                      "gp3",
```

Comment:
Lets move these into the openapi schema too 🙏

Reply:
Sounds good. I think I need to figure out how to make it so.

Reply:
Looking at the code I'm actually not sure if we can set these or `reclaimPolicy` and `volumeBindingMode`.

By default CAREN doesn't enable any addons and its always opt-in, so we don't have an opportunity in code to set these values using the API.

Maybe we need to rethink that and do enable all of these by default, using the API and then if you don't want any of them you can remove it from the Cluster object?

cc @jimmidyson I would like your thoughts too.

Reply:
Lets come back to this question and keep this PR as is since it would require broader changes.

Reply:
Are these defaults generally applicable to all storage classes? If so, I think we could set them as defaults in the OpenAPI schema for `StorageClassConfig`. Need to play with this though to see if it works like that 😅

Reply:
You said in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/10#discussion_r1543128754 that these are only applicable to AWS EBS CSI but it looks like in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/7/files#diff-e20a2f6fbb5d47965335a767f8c2681519e08dbc0eae067c019f69b655fded8aR162-R165 you're applying them generally to any provider?

Reply:
@faiq did you see Jimmi's last comment, I think its the values are always being applied here?

Reply:
yeah, fixed after writing some tests